### PR TITLE
Fix connections getting into a bad state when a task is cancelled

### PR DESCRIPTION
**Pull request by dswillr**
Resolves https://github.com/encode/databases/issues/455

### CODE REVIEW

File: tests/test_databases.py [hunk 1]
```diff
@@ -303,24 +303,26 @@ async def test_ddl_queries(database_url):
             await database.execute(query)
 
 
+class DBException(Exception):
+    pass
```
Comment (aminalaee) [inline tests/test_databases.py:307]: Maybe we don't need this?

File: tests/test_databases.py [hunk 2]
```diff
@@ -303,24 +303,26 @@ async def test_ddl_queries(database_url):
...
+
+
+@pytest.mark.parametrize("exception", [DBException, asyncio.CancelledError])
```
Comment (aminalaee) [inline tests/test_databases.py:310]: Could we do this?

```suggestion
@pytest.mark.parametrize("exception", [Exception, asyncio.CancelledError])
```

**Review by aminalaee — COMMENTED** [review]
Thank you for the PR. Looks reasonable to me.

File: tests/test_databases.py (hunk 1 above)
Comment (dswillr) [reply tests/test_databases.py:307]: I just kept what we had before, you're right we could remove that!

Comment (dswillr) [reply tests/test_databases.py]: I've applied your patch & removed the redundant class

Comment (aminalaee) [reply tests/test_databases.py]: Great, thanks.

**Review by aminalaee — APPROVED** [review]
🎉

**Comment by kevburnsjr** [comment]
Any chance you might tag a new release so we can start using this fix?

**Comment by aminalaee** [comment]
It will be in version 0.6.0.

**Pull request merged**